### ISSUE TITLE
lcd_rotate=2 not honored

### ISSUE THREAD

**Issue by Hixie** [ardera/raspberrypi-fast-ts#3]
I have lcd_rotate=2 set in my bootconfig.txt, but this driver doesn't seem to honour it, resulting in the touch coordinate space being flipped relative to the rendering coordinate space. Not sure exactly how the driver is supposed to know this or how to fix it? If you have any ideas then I'm happy to try to submit a patch myself.

**Comment by Hixie** [comment]
...or maybe it is honoured, but the code flips it? I see hflip and vflip are always set to true in the driver...

**Comment by ardera** [comment]
hflip and vflip always default to true (they do too in the kernel driver. this code is actually very similiar to the kernel driver) Then, the kernel driver sets any of those to flips to false, depending on the rotation.

I think this is best implemented using a commandline switch, since it's hard to parse `/boot/config.txt` reliably.

**Comment by ardera** [comment]
Btw, if you use this in combination with flutter-pi, you can also omit the `lcd_rotate` in `/boot/config.txt` and instead specify `--rotation 180` or `--orientation landscape_right` as a cmdline argument to `flutter-pi`.

**Comment by Hixie** [comment]
I can't omit the lcd_rotate in /boot/config.txt because then the console will be upside down. :-)

How does the kernel driver do this?

**Comment by ardera** [comment]
I'm not entirely sure, either the kernel gets its `hflip` and `vflip` from the device tree, which is provided by the bootloader (firmware) I think, or the kernel doesn't get proper `hflip` and `vflip` at all and the firmware does the flipping instead. (Since all touch events pass through the Raspberry Pi firmware running on the VideoCore before they reach the kernel)

If it's the first alternative, you can get the `hflip` and `vflip` in use by querying the device tree at runtime (maybe through some sysfs magic). Though I've tried to do that already, I dumped the whole device tree at runtime but I couldn't find anything matching `hflip` or `vflip`. Maybe it's only present if `lcd_rotate` is specified in the `config.txt`, or maybe these properties are not present at all when the default touchscreen driver is disabled (`disable_touchscreen` in `config.txt`)

**Comment by ardera** [comment]
this is now supported, see the readme
